src/Reflex/Host/Basic.hs

```diff
@@ -87,109 +86,142 @@ instance ReflexHost t => MonadSample t (BasicGuest t m) where
+
   {-# INLINABLE performEvent_ #-}
   performEvent_ = BasicGuest . lift . lift . performEvent_
+
   {-# INLINABLE performEvent #-}
   performEvent = BasicGuest . lift . lift . performEvent
 
-instance (Reflex t, ReflexHost t, Ref m ~ Ref IO, MonadHold t m, PrimMonad (HostFrame t)) => Adjustable t (BasicGuest t m) where
+instance
+  ( ReflexHost t
+  , Ref m ~ Ref IO
+  , MonadHold t m
+  , PrimMonad (HostFrame t)
+  ) => Adjustable t (BasicGuest t m) where
+
   {-# INLINABLE runWithReplace #-}
-  runWithReplace a0 a' =
-    BasicGuest $ runWithReplace (unBasicGuest a0) (fmap unBasicGuest a')
+  runWithReplace a0 a' = BasicGuest $
+    runWithReplace (unBasicGuest a0) (fmap unBasicGuest a')
+
   {-# INLINABLE traverseIntMapWithKeyWithAdjust #-}
-  traverseIntMapWithKeyWithAdjust f dm0 dm' = do
-    BasicGuest $ traverseIntMapWithKeyWithAdjust (\k v -> unBasicGuest (f k v)) dm0 dm'
+  traverseIntMapWithKeyWithAdjust f dm0 dm' = BasicGuest $
+    traverseIntMapWithKeyWithAdjust (\k v -> unBasicGuest (f k v)) dm0 dm'
+
   {-# INLINABLE traverseDMapWithKeyWithAdjust #-}
-  traverseDMapWithKeyWithAdjust f dm0 dm' = do
-    BasicGuest $ traverseDMapWithKeyWithAdjust (\k v -> unBasicGuest (f k v)) dm0 dm'
-  {-# INLINABLE traverseDMapWithKeyWithAdjustWithMove #-}
-  traverseDMapWithKeyWithAdjustWithMove f dm0 dm' = do
-    BasicGuest $ traverseDMapWithKeyWithAdjustWithMove (\k v -> unBasicGuest (f k v)) dm0 dm'
+  traverseDMapWithKeyWithAdjust f dm0 dm' = BasicGuest $
+    traverseDMapWithKeyWithAdjust (\k v -> unBasicGuest (f k v)) dm0 dm'
 
+  {-# INLINABLE traverseDMapWithKeyWithAdjustWithMove #-}
+  traverseDMapWithKeyWithAdjustWithMove f dm0 dm' = BasicGuest $
+    traverseDMapWithKeyWithAdjustWithMove (\k v -> unBasicGuest (f k v)) dm0 dm'
 
-instance (ReflexHost t) => NotReady t (BasicGuest t m) where
+instance ReflexHost t => NotReady t (BasicGuest t m) where
   {-# INLINABLE notReadyUntil #-}
   notReadyUntil _ = pure ()
+
   {-# INLINABLE notReady #-}
   notReady = pure ()
 
--- | Run a 'BasicGuest' without a quit 'Event'
-basicHostForever :: (forall t m. BasicGuestConstraints t m => BasicGuest t m a)
-                 -> IO a
-basicHostForever guest = basicHostWithQuit $ (\x -> (x, never)) <$> guest
+-- | Run a 'BasicGuest' without a quit 'Event'.
+basicHostForever
+  :: (forall t m. BasicGuestConstraints t m => BasicGuest t m a)
+  -> IO a
+basicHostForever guest = basicHostWithQuit $ guest <&> (,never)
 
--- | Run a 'BasicGuest'
+-- | Run a 'BasicGuest'.
 --
 -- The program will exit when the 'Event' returned by the 'BasicGuest' fires
-basicHostWithQuit :: (forall t m. BasicGuestConstraints t m => BasicGuest t m (a, Event t ()))
-                  -> IO a
+basicHostWithQuit
+  :: (forall t m. BasicGuestConstraints t m => BasicGuest t m (a, Event t ()))
+  -> IO a
 basicHostWithQuit (BasicGuest guest) = do
-  events <- liftIO newChan
+  performEventChan <- newChan
+  (postBuild, postBuildTriggerRef) <- runSpiderHost newEventWithTriggerRef
```

Comment:
I have a preference for using a single `runSpiderHost` for these calls.  It's a pretty small thing overall, but it'll stop me from having to load the differences between doing three things wrapped in a single `runSpiderHost` vs diong three things that are each wrapped in a `runSpiderHost`.

Reply:
```haskell
-- | Run an action affecting the global Spider timeline; this will be guarded by
-- a mutex for that timeline
runSpiderHost :: SpiderHost Global a -> IO a
runSpiderHost (SpiderHost a) = a
```

I'm not sure how the docstring is enforced, but the difference seems to be "none".

If we do the forking-style host, we'll need to do two `runSpiderHost` calls: one to construct the network and one to run it. This is unavoidable because we have to `forkIO` after construction but before looping.

Reply:
Actually this is wrong, we can `liftIO $ forkIO loop`.